pyquil/_parser/PyQuilListener.py

```diff
@@ -461,6 +564,14 @@ def _expression(expression):
     raise RuntimeError("Unexpected expression type:" + expression.getText())
 
 
+def _named_parameters(params):
+    ret = dict()
+    for param in params:
+        name = param.colonTerminatedName().getText()[:-1]
+        expr = _expression(param.expression())
+        ret[name] = expr
+
```

Comment:
`return ret`?

Or if you want to be :cool:

```python
return {param.colonTerminatedName().getText()[:-1] : _expression(param.expression()) for param in params}
```

Oh wait, no, that's hideous. Yeah your version is better.